src/anomalib/models/image/dinomaly/components/training_utils.py

```diff
@@ -0,0 +1,336 @@
+# Copyright (C) 2025 Intel Corporation
```

Comment:
Would it be an idea to split this module to two under `components` directory:
`loss.py` and `optimizer.py`? it would be easier in terms of navigation for the user/reader?